pkg/southbound/gnmi/conn_manager.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2021-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gnmi
+
+import (
+	"context"
+	"sync"
+
+	topoapi "github.com/onosproject/onos-api/go/onos/topo"
+
+	"github.com/onosproject/onos-lib-go/pkg/errors"
+)
+
+// ConnManager gNMI connection manager
+type ConnManager interface {
+	Get(ctx context.Context, id ConnID) (Conn, error)
+	List(ctx context.Context) ([]Conn, error)
+	Watch(ctx context.Context, ch chan<- Conn) error
+	Connect(target *topoapi.Object) (Conn, error)
+	Remove(connID ConnID) error
```

Comment:
I don't think we need a public `Remove` method. The addition/removal of the connection should be internal to the manager since the manager needs to monitor the connection state for changes, e.g.

```go
func (c *connManager) Connect(ctx context.Context, target *topoapi.Object) (Conn, error) {
    c.mu.Lock()
    defer c.mu.Unlock()
    if conn, ok := c.conns[target.ID]; ok {
        return conn, nil
    }

    clientConn, err := grpc.DialContext(ctx, ...)
    if err != nil {
        return nil, err
    }

    conn := NewConn(clientConn)
    c.conns[target.ID] = conn

    go func() {
        state := clientConn.GetState()
        for clientConn.WaitForStateChange(ctx, state) {
            state = clientConn.GetState()
            if state == connectivity.Shutdown {
                break
            }
        }

        c.mu.Lock()
        defer c.mu.Unlock()
        delete(c.conns, target.ID)
    }()
    return conn, nil
}
```